src/traversals/dfs.jl

```diff
@@ -5,12 +5,36 @@
     is_cyclic(g)
 
 Return `true` if graph `g` contains a cycle.
-
 ### Implementation Notes
-Uses DFS.
+The algorithm uses a DFS.
 """
 function is_cyclic end
-@traitfn is_cyclic(g::::(!IsDirected)) = ne(g) > 0
+@enum Vertex_state unvisited visited
+@traitfn function is_cyclic(g::AG::(!IsDirected)) where {T, AG<:AbstractGraph{T}}
+    vertex_state = Vector{Vertex_state}(undef, nv(g)) 
+    fill!(vertex_state,unvisited)
+    parent = zeros(nv(g))
```

Comment:
```suggestion
    parent = zeros(T, nv(g))
```
otherwise `parent` is a vector of `Float64`, which does not make much sense. It currently works thanks to int->float and float->int conversions, but those are unnecessary